src/Button/RedliningButton/RedliningButton.spec.jsx

```diff
@@ -0,0 +1,431 @@
+        drawType: 'invalid'
+      });
+
+      expect(consoleSpy).toHaveBeenCalledTimes(1);
+      expect(consoleSpy).toHaveBeenCalledWith(
+        expect.stringContaining('Warning: Failed prop type')
+      );
+
+      consoleSpy.mockReset();
+      consoleSpy.mockRestore();
+    });
+  });
+
+  describe('#Static methods', () => {
+
+    describe('#onToggle', () => {
+
+      it ('calls a createDrawInteraction method if button was pressed and valid drawType was provided', () => {
+        const wrapper = setupWrapper();
+        const createDrawInteraction = jest.spyOn(wrapper.instance(), 'createDrawInteraction');
+
+        wrapper.instance().onToggle(true);
+        expect(createDrawInteraction).toHaveBeenCalledTimes(1);
+
+        createDrawInteraction.mockReset();
+        createDrawInteraction.mockRestore();
+      });
+
+      it ('calls a createSelectOrModifyInteraction method if button was pressed and valid editType was provided', () => {
+        const wrapper = setupWrapper();
+        wrapper.setProps({
+          drawType: null,
+          editType: 'Edit'
+        });
+        const createSelectOrModifyInteraction = jest.spyOn(wrapper.instance(), 'createSelectOrModifyInteraction');
+
+        wrapper.instance().onToggle(true);
+        expect(createSelectOrModifyInteraction).toHaveBeenCalledTimes(1);
+
+        createSelectOrModifyInteraction.mockReset();
+        createSelectOrModifyInteraction.mockRestore();
+      });
+
+      it ('removes all draw/select interactions created by component from the map if the button was untoggled', () => {
+        const wrapper = setupWrapper();
+
+        const mockInteraction = getMockDrawPointInteraction();
+
+        const defaultMapInteractionsLength = map.getInteractions().getArray().length;
+        map.addInteraction(getMockDrawPointInteraction());
+        map.on('pointermove', wrapper.instance().onPointerMove);
+
+        wrapper.setState({
+          interaction: [mockInteraction]
+        });
+
+        expect(map.getInteractions().getArray().length).toBe(defaultMapInteractionsLength + 1);
+        expect(map.listeners_.pointermove).toBeDefined();
+
+        wrapper.instance().onToggle(false);
+
+        expect(map.listeners_.pointermove).toBeUndefined();
+      });
+
+      it ('unregisters `add` listener on redlining feature collection if drawType is Text and button was untoggled', () => {
+        const wrapper = setupWrapper();
+        wrapper.setProps({
+          drawType: 'Text'
+        });
+        const mockInteraction = getMockDrawPointInteraction();
+
+        const instance = wrapper.instance();
+        instance.createRedliningLayer();
+        instance._redliningFeatures =
+          wrapper.state().redliningLayer.getSource().getFeaturesCollection();
+
+        map.addInteraction(mockInteraction);
+        instance._redliningFeatures.on('add', wrapper.instance().handleTextAdding);
+
+        expect(instance._redliningFeatures.listeners_.add.length).toBe(2);
```

Comment:
This and the lines below where we use the private property `.listeners_` are a bit fragile. Right now I don't have an alternative, but at least a comment would be nice.

Reply:
comment added in 3a0a656